nomos-http-api/src/http/cl.rs

```diff
@@ -0,0 +1,68 @@
+use nomos_mempool::{
+    backend::mockpool::MockPool,
+    network::adapters::libp2p::Libp2pAdapter,
+    openapi::{MempoolMetrics, Status},
+    Transaction as TxDiscriminant,
+    MempoolMsg, MempoolService,
+};
+use tokio::sync::oneshot;
+
+
+use bytes::Bytes;
+use nomos_core::tx::{Transaction, TransactionHasher};
+use serde::{Deserialize, Serialize};
+use std::hash::Hash;
+
+#[derive(Clone, Debug, Serialize, Deserialize, Hash, PartialEq, Eq)]
+pub struct Tx(pub String);
```

Comment:
shouldn't we reuse the definitions from nomos-core?

Reply:
This `Tx` comes from `nomos-node`, I copied it from there. I am not sure if we want to use the `Tx` from `nomos-core` or do we want to make it generic over the `Transaction` trait in this service?

Reply:
uhm, At some point we used to have a custom Tx on nomos node binary if that is what @al8n refers. But yeah, the http-api should use generic types if possible. Otherwise just the types to be used by the nomos node itself. The second option should be ok imo as this service is completely ad-hoc to be used in the nomos binary. And **no way types should be in this module**/